cleanup.go

```diff
@@ -2,27 +2,66 @@ package main
 
 import (
 	"context"
-	"errors"
+	"fmt"
+	"os"
+	"time"
 
 	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/defaults"
 	"github.com/containerd/containerd/errdefs"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
 )
 
-func cleanup(ctx context.Context, container containerd.Container) error {
+func cleanup(ctx context.Context, id string) error {
+	client, err := containerd.New(
+		defaults.DefaultAddress,
+		containerd.WithDefaultRuntime("io.containerd.process.v1"),
+	)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	ctx, cancel := context.WithTimeout(ctx, 20*time.Second)
+	defer cancel()
+
+	container, err := client.LoadContainer(ctx, id)
+	if err != nil {
+		if !errdefs.IsNotFound(err) {
+			return err
+		}
+		// container does not exist so there is nothing for us to do
+		return nil
+	}
 	task, err := container.Task(ctx, nil)
 	if err != nil {
 		if errdefs.IsNotFound(err) {
+			// there is no running task, nothing to kill
 			return nil
 		}
 		return err
 	}
-	status, err := task.Status(ctx)
+	wait, err := task.Wait(ctx)
 	if err != nil {
+		task.Kill(ctx, unix.SIGKILL)
+		_, err = task.Delete(ctx)
 		return err
 	}
-	if status.Status != containerd.Stopped {
-		return errors.New("unable to start running container")
+	if err := task.Kill(ctx, unix.SIGKILL); err != nil {
+		fmt.Fprintf(os.Stderr, err.Error())
 	}
+	<-wait
```

Comment:
'nit - perhaps it's pedantic, but wondering if this could be hardened with:
```
select {
case <- wait
case <- ctx.Done():
}
```

Reply:
actually, since we pass the context to Wait() it will automatically close the connection as well as the channel on a timeout